src/expr/expr_node.rs

```diff
@@ -82,6 +90,59 @@ impl Expr {
         }
     }
 
+    pub fn slice<E, R: RangeBounds<E>>(
+        self,
+        range: impl RangeInto<E, R, (Bound<Expr>, Bound<Expr>)>,
+    ) -> Expr {
+        Expr {
+            node: Box::new(SliceNode {
+                string: self,
+                range: range.range_into(),
+            }),
+        }
+    }
+
+    pub fn revcomp_rna(self) -> Expr {
```

Comment:
For RNA alphabet support, I think its best if we chose to add it everywhere or nowhere right now. Since its very rare, I'm leaning towards not having it for now. In the future if different alphabets (methylation is another one) are needed, then we can add a more general solution (maybe an enum indicating the alphabet).